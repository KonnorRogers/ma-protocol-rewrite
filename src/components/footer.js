import React from "react"
import { useStaticQuery, graphql } from "gatsby"

import { css } from "@emotion/core"
import Screen from "../utils/screen.js"

const Footer = () => {
  const data = useStaticQuery(graphql`
    query SiteTitleQuery {
      site {
        siteMetadata {
          author
        }
      }
    }
  `)

  const footerCSS = css`
    margin: 0;
    padding: 0.5rem;
    font-size: small;
    text-align: center;
    background: rgba(0, 0, 0, 0.07);

    ${Screen.small} {
      display: flex;
      justify-content: center;
      font-size: medium;
    }

    ${Screen.medium} {
      font-size: large;
      padding: 0.25rem;
    }
  `

  const gatsbySpanCSS = css`
    padding: 0 0.25rem;
    margin: 0 0.25rem;
    color: var(--gatsby-purple);

    &:hover {
      background: var(--gatsby-purple);
      color: white;
    }
  `

  const authorCSS = css`
    display: block;
    ${Screen.small} {
      &::before {
        content: " | ";
        margin: 1px;
        padding: 1px;
      }
    }
  `

  return (
    <>
      <footer css={footerCSS}>
        Â© {new Date().getFullYear()}
        &nbsp; | &nbsp; Built with
        <a href="https://www.gatsbyjs.org" css={gatsbySpanCSS}>
          Gatsby
        </a>
        <span css={authorCSS}>
          Created and maintained by {data.site.siteMetadata.author}
        </span>
      </footer>
    </>
  )
}

export default Footer
