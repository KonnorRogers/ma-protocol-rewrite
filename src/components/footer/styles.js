import { css } from "@emotion/core"
import Screen from "../../utils/screen.js"

export const footer = css`
  margin: 0;
  padding: 0.5rem;
  font-size: small;
  text-align: center;
  background: rgba(0, 0, 0, 0.07);

  ${Screen.small} {
    display: flex;
    justify-content: center;
    font-size: medium;
  }

  ${Screen.medium} {
    font-size: large;
    padding: 0.25rem;
  }
`

export const gatsbyLink = css`
  padding: 0 0.25rem;
  margin: 0 0.25rem;
  color: var(--gatsby-purple);

  &:hover {
    background: var(--gatsby-purple);
    color: white;
  }
`

export const author = css`
  display: block;
  ${Screen.small} {
    &::before {
      content: " | ";
      margin: 1px;
      padding: 1px;
    }
  }
`
