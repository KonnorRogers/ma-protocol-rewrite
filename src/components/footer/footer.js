import React from "react"
import { useStaticQuery, graphql } from "gatsby"

import * as styles from "./styles.js"

export const PureFooter = ({ data }) => (
  <>
    <footer css={styles.footer}>
      Â© {new Date().getFullYear()}
      &nbsp; | &nbsp; Built with
      <a href="https://www.gatsbyjs.org" css={styles.gatsbyLink}>
        Gatsby
      </a>
      <span css={styles.author}>
        Created and maintained by {data.site.siteMetadata.author}
      </span>
    </footer>
  </>
)

export const Footer = props => {
  const data = useStaticQuery(graphql`
    query SiteTitleQuery {
      site {
        siteMetadata {
          author
        }
      }
    }
  `)

  return <PureFooter {...props} data={data} />
}

export default Footer
