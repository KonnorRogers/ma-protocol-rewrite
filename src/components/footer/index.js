import React from "react"
import { useStaticQuery, graphql } from "gatsby"
import PropTypes from "prop-types"

import * as styles from "./styles.js"

export const PureFooter = ({ data }) => (
  <>
    <footer css={styles.footer}>
      Â© {new Date().getFullYear()}
      &nbsp; | &nbsp; Built with
      <a href="https://www.gatsbyjs.org" css={styles.gatsbyLink}>
        Gatsby
      </a>
      <span css={styles.author}>
        Created and maintained by {data.site.siteMetadata.author}
      </span>
    </footer>
  </>
)

/**
To ensure the author is pulled in correctly
data: { site: { siteMetadata: { author: 'string' } } }
should be how the graphql queries returns the data
*/
PureFooter.propTypes = {
  data: PropTypes.shape({
    site: PropTypes.shape({
      siteMetadata: PropTypes.shape({
        author: PropTypes.string,
      }),
    }),
  }),
}

export const Footer = props => {
  const data = useStaticQuery(graphql`
    query SiteTitleQuery {
      site {
        siteMetadata {
          author
        }
      }
    }
  `)

  return <PureFooter {...props} data={data} />
}

export default Footer
