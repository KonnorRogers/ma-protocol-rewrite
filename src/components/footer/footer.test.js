import React from "react"
import { render } from "@testing-library/react"
import { PureFooter as Footer } from "./footer.js"

test("Renders without error", () => {
  const data = {
    site: {
      siteMetadata: {
        author: "Konnor Rogers",
      },
    },
  }
  const tree = render(<Footer data={data} />)
  expect(tree).toMatchSnapshot()
})
