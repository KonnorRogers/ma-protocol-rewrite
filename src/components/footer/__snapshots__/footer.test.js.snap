// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-2 {
  margin: 0;
  padding: 0.5rem;
  font-size: small;
  text-align: center;
  background: rgba(0,0,0,0.07);
}

@media only screen and (min-width:640px) {
  .emotion-2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    font-size: medium;
  }
}

@media only screen and (min-width:768px) {
  .emotion-2 {
    font-size: large;
    padding: 0.25rem;
  }
}

.emotion-0 {
  padding: 0 0.25rem;
  margin: 0 0.25rem;
  color: var(--gatsby-purple);
}

.emotion-0:hover {
  background: var(--gatsby-purple);
  color: white;
}

.emotion-1 {
  display: block;
}

@media only screen and (min-width:640px) {
  .emotion-1::before {
    content: " | ";
    margin: 1px;
    padding: 1px;
  }
}

.emotion-2 {
  margin: 0;
  padding: 0.5rem;
  font-size: small;
  text-align: center;
  background: rgba(0,0,0,0.07);
}

@media only screen and (min-width:640px) {
  .emotion-2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    font-size: medium;
  }
}

@media only screen and (min-width:768px) {
  .emotion-2 {
    font-size: large;
    padding: 0.25rem;
  }
}

.emotion-0 {
  padding: 0 0.25rem;
  margin: 0 0.25rem;
  color: var(--gatsby-purple);
}

.emotion-0:hover {
  background: var(--gatsby-purple);
  color: white;
}

.emotion-1 {
  display: block;
}

@media only screen and (min-width:640px) {
  .emotion-1::before {
    content: " | ";
    margin: 1px;
    padding: 1px;
  }
}

<body>
    <div>
      <footer
        class="emotion-2"
      >
        © 
        2020
          |   Built with
        <a
          class="emotion-0"
          href="https://www.gatsbyjs.org"
        >
          Gatsby
        </a>
        <span
          class="emotion-1"
        >
          Created and maintained by 
          Konnor Rogers
        </span>
      </footer>
    </div>
  </body>,
  "container": .emotion-2 {
  margin: 0;
  padding: 0.5rem;
  font-size: small;
  text-align: center;
  background: rgba(0,0,0,0.07);
}

@media only screen and (min-width:640px) {
  .emotion-2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    font-size: medium;
  }
}

@media only screen and (min-width:768px) {
  .emotion-2 {
    font-size: large;
    padding: 0.25rem;
  }
}

.emotion-0 {
  padding: 0 0.25rem;
  margin: 0 0.25rem;
  color: var(--gatsby-purple);
}

.emotion-0:hover {
  background: var(--gatsby-purple);
  color: white;
}

.emotion-1 {
  display: block;
}

@media only screen and (min-width:640px) {
  .emotion-1::before {
    content: " | ";
    margin: 1px;
    padding: 1px;
  }
}

.emotion-2 {
  margin: 0;
  padding: 0.5rem;
  font-size: small;
  text-align: center;
  background: rgba(0,0,0,0.07);
}

@media only screen and (min-width:640px) {
  .emotion-2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    font-size: medium;
  }
}

@media only screen and (min-width:768px) {
  .emotion-2 {
    font-size: large;
    padding: 0.25rem;
  }
}

.emotion-0 {
  padding: 0 0.25rem;
  margin: 0 0.25rem;
  color: var(--gatsby-purple);
}

.emotion-0:hover {
  background: var(--gatsby-purple);
  color: white;
}

.emotion-1 {
  display: block;
}

@media only screen and (min-width:640px) {
  .emotion-1::before {
    content: " | ";
    margin: 1px;
    padding: 1px;
  }
}

<div>
    <footer
      class="emotion-2"
    >
      © 
      2020
        |   Built with
      <a
        class="emotion-0"
        href="https://www.gatsbyjs.org"
      >
        Gatsby
      </a>
      <span
        class="emotion-1"
      >
        Created and maintained by 
        Konnor Rogers
      </span>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
